modules/doctor/heads.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package doctor
+
+import (
+	"context"
+
+	repo_model "code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/log"
+)
+
+func synchronizeRepoHeads(ctx context.Context, logger log.Logger, autofix bool) error {
+	numRepos := 0
+	numHeadsBroken := 0
+	numDefaultBranchesBroken := 0
+	numReposUpdated := 0
+	err := iterateRepositories(ctx, func(repo *repo_model.Repository) error {
+		numRepos++
+		runOpts := &git.RunOpts{Dir: repo.RepoPath()}
+
+		_, _, defaultBranchErr := git.NewCommand(ctx, "rev-parse", repo.DefaultBranch).RunStdString(runOpts)
```

Comment:
IIRC, every user provided dynamic argument should be put after `--`?

Not a blocker, just a question.

Reply:
That's a good point, and I wonder if we have any tests that check in other places what happens if user branches start with `-` or `--`.

Reply:
Thanks @wxiaoguang, sorry I missed that

```suggestion
		_, _, defaultBranchErr := git.NewCommand(ctx, "rev-parse", "--", repo.DefaultBranch).RunStdString(runOpts)
```

Reply:
There were two places it needed to be added to, so, I added to both.